include/awkward/Iterator.h

```diff
@@ -1,33 +1,34 @@
-// BSD 3-Clause License; see https://github.com/jpivarski/awkward-1.0/blob/master/LICENSE
+// BSD 3-Clause License; see
+// https://github.com/jpivarski/awkward-1.0/blob/master/LICENSE
 
 #ifndef AWKWARD_ITERATOR_H_
 #define AWKWARD_ITERATOR_H_
 
-#include "awkward/cpu-kernels/util.h"
 #include "awkward/Content.h"
+#include "awkward/cpu-kernels/util.h"
 
 namespace awkward {
-  class Iterator {
-  public:
-    Iterator(const std::shared_ptr<Content> content)
-        : content_(content)
-        , at_(0) {
-      content.get()->check_for_iteration();
-    }
-
-    const std::shared_ptr<Content> content() const { return content_; }
-    const int64_t at() const { return at_; }
-
-    const bool isdone() const;
-    const std::shared_ptr<Content> next();
-
-    const std::string tostring_part(const std::string indent, const std::string pre, const std::string post) const;
-    const std::string tostring() const;
-
-  private:
-    const std::shared_ptr<Content> content_;
-    int64_t at_;
-  };
-}
+class Iterator {
+public:
+  Iterator(const std::shared_ptr<Content> content) : content_(content), at_(0) {
+    content.get()->check_for_iteration();
+  }
+
+  const std::shared_ptr<Content> content() const { return content_; }
+  const int64_t at() const { return at_; }
+
+  const bool isdone() const;
+  const std::shared_ptr<Content> next();
+
+  const std::string tostring_part(const std::string indent,
+                                  const std::string pre,
+                                  const std::string post) const;
```

Comment:
These kinds of linewidth reductions aren't needed because `const std::string indent, const std::string pre, const string post` is a common pattern.

Now here's something that I wish it had caught: I would have liked those to be passing constant references so that it doesn't copy strings. It doesn't matter all that much for performance (our performance efforts would be concentrated on the cpu-kernels in the deepest layer of the architecture), but I'd like to be consistent.

I've been really inconsistent about how I pass references to heavyweight (non-numeric) objects: I'd like it to be `const ObjectType& objectvalue`, but I sometimes forget the `&` and I sometimes forget the `const`. I _very rarely_ mutate objects in place and most of the methods don't change their class, so there should be `const` almost everywhere. (I wish C++ opted into mutability instead of opting into immutability, but oh well.)

Reply:
By the way, my statement of "I would have liked" doesn't mean that you need to do that sweep. Either one of us but probably me will do it and it won't take much more than an hour. I'm just saying that a more useful warning would be one that catches the lack of const-references, rather than the indentation.